### PR TITLE
Use net-http2 gem for apns auth key client

### PR DESCRIPTION
**Pull request by larskuhnt**

### CODE REVIEW

File: lib/pushmeup/apns/http2_client.rb [hunk 1]
```diff
@@ -30,71 +31,26 @@ def send_notification(device_token, bundle_identifier, message)
...
+        path = "/3/device/#{notification.device_token}"
+        request = @client.prepare_request(:post, path, headers: headers(bundle_identifier), body: notification.to_json)
+        @client.call_async(request)
+        @client.join
       end
-      yield
     end
 
-    # Close socked and ssl only if they are not nil
     def close_socket_and_ssl
```
Comment (MGPalmer) [inline lib/pushmeup/apns/http2_client.rb]: Is this method even used anymore?

File: lib/pushmeup/apns/http2_client.rb [hunk 2]
```diff
@@ -30,71 +31,26 @@ def send_notification(device_token, bundle_identifier, message)
...
-    end
-
-    def with_connection
-      if @ssl.nil? || @ssl.closed?
-        open_connection
+      notifications.each do |notification|
+        path = "/3/device/#{notification.device_token}"
+        request = @client.prepare_request(:post, path, headers: headers(bundle_identifier), body: notification.to_json)
+        @client.call_async(request)
+        @client.join
```
Comment (MGPalmer) [inline lib/pushmeup/apns/http2_client.rb]: https://github.com/ostinelli/net-http2

It looks to me as if we're missing the final close of the connection/client, and also shouldn't we "join" outside of the loop?

**Review by MGPalmer — CHANGES_REQUESTED** [review]

File: lib/pushmeup/apns/http2_client.rb (hunk 2 above)
Comment (larskuhnt) [reply lib/pushmeup/apns/http2_client.rb]: the join outside results in errors when sending multiple messages

Comment (larskuhnt) [reply lib/pushmeup/apns/http2_client.rb]: if we close it we are not able to reuse the connection, I want the caller to be responsible for closing the connection.

File: lib/pushmeup/apns/http2_client.rb (hunk 1 above)
Comment (larskuhnt) [reply lib/pushmeup/apns/http2_client.rb]: yes, because the caller is responsible for closing the connection.

**Review by MGPalmer — APPROVED** [review]

**Pull request merged**